src/js/components/xhr.js

```diff
@@ -0,0 +1,48 @@
+export function xhrPromise(options){
+    return new Promise((resolve, reject)=>{
+        try {
+            let xhr = new XMLHttpRequest();
+            xhr.open(options.method, options.url, true);
+            xhr.responseType = options.responseType || "json";
+            xhr.onload = function(){
+                if (this.status >= 200 && this.status < 300) {
```

Comment:
Hmm,  with this.status >= 200 && this.status < 300 we are checking if status is between 200 and 299. what you are suggesting is any number less than 299.